### DISCUSSION TITLE
templating: container command and args using `string` not `[]string`

### DISCUSSION THREAD

**Discussion by shenshanyoumu** [argoproj/argo-workflows#13124] [Q&A] [answered]
`command: '{{inputs.parameters.k8s_cmds}}'`, is like `'['/bin/bash','-c']'`

```
cannot unmarshal string into Go struct field Container.spec.templates.container.command of type []string
```

**Comment by agilgur5** [comment, marked as answer]
The error is correct. You gave it a string input, but `command` accepts a string array.

`command: '['/bin/bash','-c']'` is invalid, note the quotes _around_ the array.
You would want `command: ['/bin/bash','-c']`.

Argo can't do that transformation for you; ultimately a template is still a YAML string and Argo would not know what you want to convert.

You could instead do, for instance:
```yaml
command:
  - '{{inputs.parameters.k8s_cmd_part1}}'
  - '{{inputs.parameters.k8s_cmd_part2}}'
```

**Reply by shenshanyoumu** [reply]
yes I know it's hard to adapt. but variables `{{inputs.parameters.k8s_cmds}}` I don't know details. so can't expand like
```yaml
command:
  - '{{inputs.parameters.k8s_cmd_part1}}'
  - '{{inputs.parameters.k8s_cmd_part2}}'
```

**Reply by agilgur5** [reply]
You need to change your inputs or otherwise parse the string yourself. Your current inputs are invalid.

I'm not sure what else I can tell you. That's not a bug, your inputs don't match the k8s schema.
